fdbserver/include/fdbserver/workloads/MockDDTest.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * MockDDTest.g
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef FOUNDATIONDB_MOCKDDTEST_H
+#define FOUNDATIONDB_MOCKDDTEST_H
+
+#include "fdbserver/workloads/workloads.actor.h"
+#include "fdbserver/DDSharedContext.h"
+#include "fdbserver/DDTxnProcessor.h"
+#include "fdbserver/MoveKeys.actor.h"
+#include "fdbclient/StorageServerInterface.h"
+
+// other Mock DD workload can derive from this class to use the common settings
+struct MockDDTestWorkload : public TestWorkload {
+	bool enabled;
```

Comment:
I think it would be more maintainable to add a protected virtual `isEnabled` method. As currently implemented, if a new class inherits from `MockDDTestWorkload` but forgets to update `enabled` in the constructor, the memory will be uninitialized.

Reply:
The MockDDTestWorkload constructor always give an value to `enabled`